core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -122,5 +161,29 @@ class DynamicConfigManager(private val zkUtils: ZkUtils,
    */
   def startup(): Unit = {
     configChangeListener.init()
+
+    // Apply all existing client/user configs to the ClientIdConfigHandler/UserConfigHandler to bootstrap the overrides
+    configHandlers.foreach {
+      case (ConfigType.Client, handler) =>
+          AdminUtils.fetchAllEntityConfigs(zkUtils, ConfigType.Client).foreach {
+            case (clientId, properties) => handler.processConfigChanges(clientId, properties)
+          }
+      case (ConfigType.User, handler) =>
+          AdminUtils.fetchAllEntityConfigs(zkUtils, ConfigType.User).foreach {
+            case (sanitizedUser, properties) => handler.processConfigChanges(sanitizedUser, properties)
+          }
+          AdminUtils.fetchAllChildEntityConfigs(zkUtils, ConfigType.User, ConfigType.Client).foreach {
+            case (sanitizedUserClientId, properties) => handler.processConfigChanges(sanitizedUserClientId, properties)
+          }
+      case (ConfigType.Broker, handler) =>
+          AdminUtils.fetchAllEntityConfigs(zkUtils, ConfigType.Broker).foreach {
+            case (broker, properties) => handler.processConfigChanges(broker, properties)
+          }
+      case (ConfigType.Topic, handler) =>
+          AdminUtils.fetchAllEntityConfigs(zkUtils, ConfigType.Topic).foreach {
+            case (topic, properties) => handler.processConfigChanges(topic, properties)
+          }
+      case _ =>
```

Comment:
It seems that we can use the catch all case to handle all client, broker and topic?